src/view/DashboardView.java

```diff
@@ -0,0 +1,202 @@
+                    }
+                } else {
+                    JOptionPane.showMessageDialog(null, "Please enter a description for your project");
+                }
+            } else {
+                JOptionPane.showMessageDialog(null, "Please enter a title for your project");
+            }
+        });
+
+        popupPanel.add(nameLabel);
+        popupPanel.add(nameField);
+        popupPanel.add(descriptionLabel);
+        popupPanel.add(new JScrollPane(descriptionArea));
+        popupPanel.add(deadlineLabel);
+        popupPanel.add(deadlineField);
+        popupPanel.add(new JLabel());
+        popupPanel.add(submitButton);
+
+        popupFrame.add(popupPanel);
+        popupFrame.setVisible(true);
+    }
+
+    private class ProjectPanel extends JPanel {
+        private final ProjectData projectData;
+        private static final int PANEL_MIN_MAX_HEIGHT = 100;
+        private static final int ARC_SIZE = 20;
+
+        public ProjectPanel(ProjectData projectData) {
+            this.projectData = projectData;
+            setMinimumSize(new Dimension((int) (DashboardView.this.getWidth() * 0.95), PANEL_MIN_MAX_HEIGHT));
+            setMaximumSize(new Dimension((int) (DashboardView.this.getWidth() * 0.95), PANEL_MIN_MAX_HEIGHT));
+            setPreferredSize(new Dimension((int) (DashboardView.this.getWidth() * 0.95), PANEL_MIN_MAX_HEIGHT));
+            setBorder(BorderFactory.createEmptyBorder(20, 10, 0, 10));
+
+            addMouseListener(new java.awt.event.MouseAdapter() {
+                public void mouseClicked(java.awt.event.MouseEvent evt) {
+                    //TODO: PLACEHOLDER FOR VEDANTS PROJECT VIEW POPUP
+                    JOptionPane.showMessageDialog(null, "Clicked on project: " + projectData.getProjectTitle());
+                }
+            });
+        }
+
+        @Override
+        protected void paintComponent(Graphics g) {
+            super.paintComponent(g);
+            Graphics2D g2d = (Graphics2D) g.create();
+            g2d.setColor(Color.LIGHT_GRAY);
+            g2d.fillRoundRect(0, 0, getWidth(), getHeight(), ARC_SIZE, ARC_SIZE);
+            g2d.setColor(Color.BLACK);
+            g2d.setFont(new Font("Arial", Font.BOLD, 16));
+            FontMetrics metrics = g2d.getFontMetrics();
+            int textWidth = metrics.stringWidth(projectData.getProjectTitle());
+            int x = (getWidth() - textWidth) / 2;
+            int y = getHeight() / 2 + metrics.getHeight() / 4;
+            g2d.drawString(projectData.getProjectTitle(), x, y);
+            g2d.dispose();
+        }
+    }
+
+    public void propertyChange(PropertyChangeEvent evt) {
+        DashboardState state = (DashboardState) evt.getNewValue();
+
+        if (state.getAddProjectError() != null) {
+            JOptionPane.showMessageDialog(this, state.getAddProjectError());
+            state.setAddProjectError(null);
```

Comment:
SUGGESTION: Instead of resetting the state to null here, maybe do it in AddProjectPresenter.prepareSuccessView()?

Reply:
True!